From 43d4cec7abc8267b65cb7390246d175ac9b77505 Mon Sep 17 00:00:00 2001
From: Karthik M <quic_karm@quicinc.com>
Date: Fri, 10 Feb 2023 12:36:05 +0530
Subject: [PATCH] wifi: ath11k: Add dummy link_id support for fwtest

cfg80211 and mac80211 passes link_id in fw_test
command to support MLO architecture.
Add dummy link_id paramter for compatability.

Signed-off-by: Karthik M <quic_karm@quicinc.com>
---
 drivers/net/wireless/ath/ath11k/testmode.c | 3 ++-
 drivers/net/wireless/ath/ath11k/testmode.h | 4 ++--
 2 files changed, 4 insertions(+), 3 deletions(-)

--- a/drivers/net/wireless/ath/ath11k/testmode.c
+++ b/drivers/net/wireless/ath/ath11k/testmode.c
@@ -513,7 +513,7 @@ out:
 }
 
 int ath11k_tm_cmd(struct ieee80211_hw *hw, struct ieee80211_vif *vif,
-		  void *data, int len)
+		  u8 link_id, void *data, int len)
 {
 	struct ath11k *ar = hw->priv;
 	struct nlattr *tb[ATH11K_TM_ATTR_MAX + 1];
@@ -521,6 +521,7 @@ int ath11k_tm_cmd(struct ieee80211_hw *h
 
 	ret = nla_parse(tb, ATH11K_TM_ATTR_MAX, data, len, ath11k_tm_policy,
 			NULL);
+
 	if (ret)
 		return ret;
 
--- a/drivers/net/wireless/ath/ath11k/testmode.h
+++ b/drivers/net/wireless/ath/ath11k/testmode.h
@@ -10,7 +10,7 @@
 
 void ath11k_tm_wmi_event(struct ath11k_base *ab, u32 cmd_id, struct sk_buff *skb);
 int ath11k_tm_cmd(struct ieee80211_hw *hw, struct ieee80211_vif *vif,
-		  void *data, int len);
+		  u8 link_id, void *data, int len);
 
 void ath11k_fwlog_write(struct ath11k_base *ab,  u8 *data, int len);
 #else
@@ -22,7 +22,7 @@ static inline void ath11k_tm_wmi_event(s
 
 static inline int ath11k_tm_cmd(struct ieee80211_hw *hw,
 				struct ieee80211_vif *vif,
-				void *data, int len)
+				u8 link_id, void *data, int len)
 {
 	return 0;
 }
